phpBB/posting.php

```diff
@@ -1792,23 +1812,28 @@
 $notify_checked		= (isset($notify)) ? $notify : (($mode == 'post') ? $user->data['user_notify'] : $notify_set);
 
 // Page title & action URL
-$s_action = append_sid("{$phpbb_root_path}posting.$phpEx", "mode=$mode&amp;f=$forum_id");
-$s_action .= ($topic_id) ? "&amp;t=$topic_id" : '';
-$s_action .= ($post_id) ? "&amp;p=$post_id" : '';
+$s_action = append_sid("{$phpbb_root_path}posting.$phpEx", "mode=$mode");
 
 switch ($mode)
 {
 	case 'post':
+		$s_action .= ($forum_id) ? "&amp;f=$forum_id" : '';
 		$page_title = $user->lang['POST_TOPIC'];
 	break;
 
-	case 'quote':
 	case 'reply':
+		$s_action .= ($topic_id) ? "&amp;t=$topic_id" : '';
+		$page_title = $user->lang['POST_REPLY'];
```

Comment:
New language strings should use the `lang()` function, rather than the magic array access.
And preferably even use the `$language->lang()`.

Reply:
OK

Reply:
It is not really a new language string, it is just the existing case for reply and quote being split into separate cases.  I think that it is more appropriate here to keep consistency with the rest of the file until someone decides to update all of the language string syntax.

Reply:
I guess he meant "new" as in "amended ones" in the code, since the `$language->lang()` it is already used at least in one occurence here in this file it is better to follow that example.

Reply:
Same as with the quotes: Both ways work, so follow devs preferences. If there is a practical difference between array and function usage, this should be discussed in a different issue dealing with coding style/standards but not here when it's about to do an unrelated enhancement (IMO a quite large one!) added 🙂.

Reply:
Current way is fine for this one line.